[Q]: Step14-1 미션 PR

park78951 opened the pull request:
Step14-1 미션 검토 부탁드립니다.

처음에는 data를 관리하는 module과 view를 관리하는 module을 클래스로 각각 만들고 싶었지만 데이터의 변경이 없다보니 필요성이 없어져서 *View* 를 관리하는 module만 클래스로 만들었습니다.

Carousel Header(Navigator) 부분과 Main Contents 부분의 내용을 모두 template literal을 사용해서 rendering 되도록 하였으며 template literal을 만들어주는 함수를 따로 module로 빼서 View 클래스에서 호출하는 방식으로 설계했습니다.

설계하면서 새롭게 방식을 배우고 짜다보니 재밌게 짰던것 같은데, 다른 패턴들도 계속 봐야될것 같습니다.

park78951 committed:
add blocking event function when animating

PR 머지 이후 Branch 변경 및 reabse 작업 진행.
carousel.js 파일에 animation 기능 수행 중 추가 animation 일어나지 않도록
버특 막는 기능 추가

park78951 committed:
create promise.js file

Trying to Promise-like codes to implment like Promise library to
understand Promise.
Create MyPromise class and test successful and failure async codes
and successful sync codes.

park78951 committed:
create cards.json file

create json file for cards information based on index.html.

park78951 committed:
create index.html file and change the file names

create index.html file to implement fetchJson.js with localData.json to
fetch data from json file and change name of files ( cars.json ->
localData.json, fetch.js -> fetchJson.js )

park78951 committed:
Create 2 files for JSON and data model

create data.js file to fetch data from localData.json and set is as a
state in the Data class. With this, create localData.json file to store
information about image URLs, titles and description of contents in
carousel cards.

park78951 committed:
Create viewer.js file

create viewer.js file to render on the page by using json file fetched
by data.js. It's not yet tested and needs to set dependencies with
other moudles or classes.

park78951 committed:
Create localData.json and viewer.js and modifiy app.js

create viewer.js file to render caraousel main contents using template literal and modify app.js to fetch data from localData.json asynchronously and call viewer.render method with viewer class instantiated.

park78951 committed:
Modify viewer.js and app.js, create template.js

Modify viewer.js file to get a part of header(navigator) rendered with
json file by Viewer.js and create viewer.js file only to make html
templates.

A part of change from carousel.js is to get class name 'active' for
hover in style sheet due to the reason of new render of the header items and
the class name active should be assigned for one among them.

park78951 committed:
Rename URL of fetch api in app.js

park78951 committed:
Organize codes

park78951 committed:
Modify mypromise.js of condition in resolve and reject

park78951 committed:
Merge remote-tracking branch 'upstream/park78951' into step14-1

# Conflicts:
#	index.html
#	mypromise/myPromise.js
#	src/css/style.css
#	src/js/carousel.js
#	src/scss/style.css.map
#	src/scss/style.scss

park78951 committed:
Figure out conflict after merge with step13-1

crongro commented:
'필요성이 없어져서 View 를 관리하는 module만 클래스로 만들었습니다.''

-> 스스로 판단하는 과정이 좋네요.

crongro commented on app.js:
```diff
@@ -1,7 +1,25 @@
+const headerViewer = new Viewer(config.header, template.getHeaderTemplate);
+const mainViewer = new Viewer(config.container, template.getMainTempalte);
+
+const getData = async () => 
```
getData 보다는 구체적인 이름으로 표현해야겠죠.

crongro commented on app.js:
```diff
@@ -1,7 +1,25 @@
\ No newline at end of file
+  const localData = getData();
+
+  localData
```
화면렌더링이 순차적으로 headerViewer ->  mainViewer 로 진행되는 게 맞나요? 
동시에  될 수 없는건가요?

park78951 replied:
사실 동시에 해야할지를 고민했었는데, 크게 상관이 없는 것 같아보여 가독성이 더 낫지 않을까 라고 생각해서 순차적으로 진행되도록 코드를 작성했습니다. 말씀해주셔서 다시보니 불필요하게 then 작업이 추가되고 코드가 길어지는 것 같네요.... 변경하도록 하겠습니다.

crongro replied:
순처적이면 작업이 느리죠. 
병렬로 처리되도록 해도 될 거 같아요.

park78951 replied:
오... 넵 피드백 감사합니다. 말씀해주신대로 변경했습니다.

crongro commented on src/js/helper/template.js:
```diff
@@ -0,0 +1,27 @@
+const template = {
```
template을 분리하는 것도 좋은 전략이죠.
지저분해서..

아니면 이런 구조도 좋고요.
/header/header.js
/header/headerTpl.js

/main/main.js
/main/mainTpl.js

이런식으로 디렉토리별로 template을 관리할 수도 있겠네요.
(참고)

crongro commented on src/js/viewer.js:
```diff
@@ -0,0 +1,18 @@
+class Viewer {
```
Viewer보다는  자세한  UI이름을 지어주셔야..

crongro reviewed (changes requested):
질문이 있어서 request changes 합니다.

park78951 committed:
Refactor codes in app.js and rename

1. Refactor codes to be clean: reduce 'then' chain to render mainViewer
in app.js
2. Rename function 'getData' clearly to 'getCarouselData' in app.js
3. Rename file and class name: Viewer.js and Viewer -> RenderTemplate
both

crongro commented on app.js:
```diff
@@ -1,7 +1,25 @@
+      return json;
+    })
+    .then( () => {
+      const carousel = new Carousel(config);
```
이렇게 되면 header와 main이 렌더링된 이후에, 
carousel이 되는 순서인데요. 이건 맞겠죠? 

순차적으로 진행되는 것이면  이렇게 구현하는게 맞겠죠.

crongro reviewed (approved):
머지합니다. 수고하셨어요!
